dali/test/python/test_operator_python_function.py

```diff
@@ -354,3 +354,39 @@ def test_sink():
     for file in created_files:
         os.remove(file)
     os.rmdir(SINK_PATH)
+
+
+counter = 0
+def func_with_side_effects(images):
+    global counter
+    counter = counter + 1
+
+    print('Call ' + str(counter))
+
+    return numpy.full_like(images, counter)
+
+def test_func_with_side_effects():
+    pipe_one = PythonOperatorPipeline(
+        BATCH_SIZE, NUM_WORKERS, DEVICE_ID, SEED, images_dir, func_with_side_effects)
+    pipe_two = PythonOperatorPipeline(
+        BATCH_SIZE, NUM_WORKERS, DEVICE_ID, SEED, images_dir, func_with_side_effects)
+
+    pipe_one.build()
+    pipe_two.build()
+
+    global counter
+
+    for it in range(ITERS):
+        counter = 0
+        out_one, = pipe_one.run()
+        out_two, = pipe_two.run()
+
+        print('Iter ' + str(it) + ' Len one ' + str(len(out_one)) + ' len two ' + str(len(out_two)))
+        assert counter == len(out_one) + len(out_two)
+
+        for s in range(BATCH_SIZE):
+            elem_one = out_one.at(s)[0][0][0]
+            elem_two = out_two.at(s)[0][0][0]
+
+            assert 0 < elem_one <= len(out_one)
```

Comment:
Can we put more strict constrains or there is no predictable relation between sample index and value inside it?

Reply:
I don't see any more strict constrain as we can't make any assumptions about the order of sample processing. But it is quite good test, as usually in a race condition the counter ends up being too low and it's caught by the constrain for the second pipeline's result.

Reply:
Maybe we could sort and check if everything is there, as you were doing it locally?

Reply:
Yeah it sounds better. Done.